test/NonceTracker/nonceTracker.t.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+import {Test} from "forge-std/Test.sol";
+import {NonceTracker} from "../../src/NonceTracker.sol";
+
+contract NonceTrackerTest is Test {
+    NonceTracker public nonceTracker;
+    address public account;
+    uint256 constant ACCOUNT_PK = 0xA11CE;
+
+    event NonceUsed(address indexed account, uint256 nonce);
+
+    function setUp() public {
+        nonceTracker = new NonceTracker();
+        account = vm.addr(ACCOUNT_PK);
+    }
+
+    function test_initialNonceIsZero() public {
```

Comment:
style nit: if using one test file for whole contract, test names should begin with function being tested:
```suggestion
    function test_getNextNonce_initialNonceIsZero() public {
```